src/main/java/ca/corefacility/bioinformatics/irida/ria/web/projects/dto/ProjectSampleMetadataResponse.java

```diff
@@ -7,21 +7,12 @@
  */
 public class ProjectSampleMetadataResponse {
 
-	private String messageKey;
```

Comment:
We should talk about this class.  I have set up a class call `AjaxSuccessResponse` and `AjaxErrorResponse` that both extend `AjaxResponse`.  You might need to create a new class that extends `AjaxErrorResponse` that can take a list of errors.  You should not need a `messageKey`, since from what I was reading is just a string for `success` or `error` but I might be wrong, since this should just be handles in the `ResponseEntity` status.

Reply:
Yes, please. In an earlier comment I changed it to throw an exception on error. Recently, I made a small change to only return a string to return an AjaxSuccessResponse, see f77cd807dae50ea791ad91fbc2d8084045fa0df7. Maybe I can keep track of the "errorList" in SampleMetadataStorageRow? Then I can remain on the review page and show each error on the table row?